lib/handlers/router-entities/route-caching/dynamo-route-caching-provider.ts

```diff
@@ -196,90 +181,7 @@ export class DynamoRouteCachingProvider extends IRouteCachingProvider {
-      tokenIn,
-      tokenOut,
-      chainId,
-      amount,
-      tradeType,
-      protocols,
-      currentBlockNumber,
-      optimistic,
-      partitionKey
-    )
-
-    const routesDbPromise = this.getRoutesFromRoutesDb(partitionKey, chainId, amount, currentBlockNumber, optimistic)
-
-    const [cachedRoutes, routesDb] = await Promise.all([cachedRoutesPromise, routesDbPromise])
-
-    return cachedRoutes || routesDb
-  }
-
-  private async getCachedRoutesFromCachedRoutesDb(
-    tokenIn: Token,
-    tokenOut: Token,
-    chainId: ChainId,
-    amount: CurrencyAmount<Currency>,
-    tradeType: TradeType,
-    protocols: Protocol[],
-    currentBlockNumber: number,
-    optimistic: boolean,
-    partitionKey: PairTradeTypeChainId
-  ): Promise<CachedRoutes | undefined> {
-    const cachedRoutesStrategy = this.getCachedRoutesStrategy(tokenIn, tokenOut, tradeType, chainId)
-    const cachingBucket = cachedRoutesStrategy?.getCachingBucket(amount)
-
-    if (cachingBucket) {
-      const sortKey = new ProtocolsBucketBlockNumber({
-        protocols,
-        bucket: cachingBucket.bucket,
-        blockNumber: currentBlockNumber,
-      })
-
-      try {
-        const queryParams = {
-          TableName: this.cachingRoutesTableName,
-          // Since we don't know what's the latest block that we have in cache, we make a query with a partial sort key
-          KeyConditionExpression: '#pk = :pk and begins_with(#sk, :sk)',
-          ExpressionAttributeNames: {
-            '#pk': 'pairTradeTypeChainId',
-            '#sk': 'protocolsBucketBlockNumber',
-          },
-          ExpressionAttributeValues: {
-            ':pk': partitionKey.toString(),
-            ':sk': sortKey.protocolsBucketPartialKey(),
-          },
-          ScanIndexForward: false, // Reverse order to retrieve most recent item first
-          Limit: Math.max(cachingBucket.withLastNCachedRoutes, 1),
-        }
-
-        const result = await this.ddbClient.query(queryParams).promise()
-
-        if (result.Items && result.Items.length > 0) {
-          return this.parseCachedRoutes(
-            result,
-            chainId,
-            currentBlockNumber,
-            optimistic,
-            partitionKey,
-            amount,
-            CachedRoutesSource.CachedRoutes,
-            sortKey
-          )
-        } else {
-          metric.putMetric('CachedRoutesEntriesNotFound', 1, MetricLoggerUnit.Count)
-          log.warn(`[DynamoRouteCachingProvider] No items found in the query response for ${partitionKey.toString()}`)
-        }
-      } catch (error) {
-        metric.putMetric('CachedRoutesFetchError', 1, MetricLoggerUnit.Count)
-        log.error(
-          { partitionKey, sortKey, error },
-          `[DynamoRouteCachingProvider] Error while fetching route from cache`
-        )
-      }
-    }
-
-    return undefined
+    return await this.getRoutesFromRoutesDb(partitionKey, chainId, amount, currentBlockNumber, optimistic)
```

Comment:
Maybe remove the `getRoutesFromRoutesDb` function too? since we can collapse it all down to a single function